File: data/evaluation/geeks_for_geeks_successful_test_scripts/python/DICE_THROW_PROBLEM.py
# Copyright (c) 2019-present, Facebook, Inc.
# All rights reserved.
#
# This source code is licensed under the license found in the
# LICENSE file in the root directory of this source tree.
#
def f_gold ( m , n , x ) :
    table = [ [ 0 ] * ( x + 1 ) for i in range ( n + 1 ) ]
    for j in range ( 1 , min ( m + 1 , x + 1 ) ) :
        table [ 1 ] [ j ] = 1
    for i in range ( 2 , n + 1 ) :
        for j in range ( 1 , x + 1 ) :
            for k in range ( 1 , min ( m + 1 , j ) ) :
                table [ i ] [ j ] += table [ i - 1 ] [ j - k ]
    return table [ - 1 ] [ - 1 ]


#TOFILL

if __name__ == '__main__':
    param = [
    (94,4,69,),
    (7,12,33,),
    (20,44,24,),
    (90,94,88,),
    (50,58,27,),
    (32,90,29,),
    (46,25,6,),
    (82,50,87,),
    (43,82,70,),
    (6,83,19,)
        ]
    n_success = 0
    for i, parameters_set in enumerate(param):
        if f_filled(*parameters_set) == f_gold(*parameters_set):
            n_success+=1
    print("#Results: %i, %i" % (n_success, len(param)))

File: data/evaluation/geeks_for_geeks_successful_test_scripts/java/DICE_THROW_PROBLEM.java
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

import java.util. *;
import java.util.stream.*;
import java.lang.*;
import javafx.util.Pair;
public class DICE_THROW_PROBLEM{
public static long f_gold ( int m , int n , int x ) {
  long [ ] [ ] table = new long [ n + 1 ] [ x + 1 ] ;
  for ( int j = 1 ;
  j <= m && j <= x ;
  j ++ ) table [ 1 ] [ j ] = 1 ;
  for ( int i = 2 ;
  i <= n ;
  i ++ ) {
    for ( int j = 1 ;
    j <= x ;
    j ++ ) {
      for ( int k = 1 ;
      k < j && k <= m ;
      k ++ ) table [ i ] [ j ] += table [ i - 1 ] [ j - k ] ;
    }
  }
  return table [ n ] [ x ] ;
}


//TOFILL

public static void main(String args[]) {
    int n_success = 0;
    List<Integer> param0 = new ArrayList<>();
    param0.add(94);
    param0.add(7);
    param0.add(20);
    param0.add(90);
    param0.add(50);
    param0.add(32);
    param0.add(46);
    param0.add(82);
    param0.add(43);
    param0.add(6);
    List<Integer> param1 = new ArrayList<>();
    param1.add(4);
    param1.add(12);
    param1.add(44);
    param1.add(94);
    param1.add(58);
    param1.add(90);
    param1.add(25);
    param1.add(50);
    param1.add(82);
    param1.add(83);
    List<Integer> param2 = new ArrayList<>();
    param2.add(69);
    param2.add(33);
    param2.add(24);
    param2.add(88);
    param2.add(27);
    param2.add(29);
    param2.add(6);
    param2.add(87);
    param2.add(70);
    param2.add(19);
    for(int i = 0; i < param0.size(); ++i)
    {
        if(f_filled(param0.get(i),param1.get(i),param2.get(i)) == f_gold(param0.get(i),param1.get(i),param2.get(i)))
        {
            n_success+=1;
        }
    }
    System.out.println("#Results:" + n_success + ", " + param0.size());
}
}